src/views/open_order_view.js

```diff
@@ -0,0 +1,51 @@
+import Backbone from 'backbone';
+import Quote from '../models/quote';
+import Order from '../models/order';
+import OrderList from '../collections/order_list';
+import $ from 'jquery';
+
+const OpenOrderView = Backbone.View.extend({
+  initialize(params) {
+    this.template = params.template;
+    // this.quotes = params.quotes;
+    this.model = params.model;
+    this.listenTo(this.model, 'change', this.render);
+    this.listenTo(this.model.get('quote'), 'change', this.completeOrder);
+
+  },
+  completeOrder() {
+    console.log('the complete order method is being called');
+    let quote = this.model.get('quote');
+
+    if (this.model.get('buy')) {
+      if (this.model.get('targetPrice') >= quote.get('price')) {
+        quote.buy();
+        this.model.destroy();
+        this.remove();
+      }
+    } else {
+      if (this.model.get('targetPrice') <= quote.get('price')) {
+        quote.sell();
+        this.model.destroy();
+        this.remove();
+      }
+    }
+  },
+  render() {
+    const compiledTemplate = this.template(this.model.toJSON());
+
+    this.$el.html(compiledTemplate);
+
+    return this;
+  },
+  events: {
+    'click button.btn-cancel': 'cancel',
+  },
+  cancel(event) {
+    if (confirm("Are you sure?") === true){
+      this.model.destroy();
```

Comment:
We also need to call `this.remove()` here, as we do in `completeOrder()`.

What's currently happening is that we destroy the `Order` model, which triggers `update` on the `OrderList` collection it was in, and that results in `OpenOrderListView` calling its `render()` function, which creates new `OpenOrderView` instances for each order remaining in the collection. _However_, it does not get rid of the old view instances.

As a result, when we cancel an order it is no longer listed in the open orders list but the view _does_ stick around in memory -- and it's still listening to the `Quote` model for any updates. Thus, the cancelled order could still execute after it is no longer displayed. You can repro this by creating an order and clicking cancel immediately, then waiting until the relevant quote's price shifts enough that the cancelled order would have executed. You can see in the trade history that in fact the quote does get executed even after being cancelled.